shpc/main/modules/templates/includes/load_view.lua

```diff
@@ -1,6 +1,6 @@
 local view_dir = string.gsub(myFileName():match("(.*[/])") or ".","/%a*/$","")
 local view_name = string.gsub(view_dir,".*/","")
-local view_module = '.view_module'
+local view_module = '.view_' .. view_name
```

Comment:
This gets populated into the module file before any views are created, an we cannot know in advance what the view will be called.

Reply:
Look at this (and equivalent in TCL):
```
local view_name = string.gsub(view_dir,".*/","")
```
I am getting the view_name dynamically at module load time!! :-)

Reply:
Okay so what if some other module file is added there? Why do we need this approach again?

I’m off to bed but I hope @georgiastuart can test this out!

Reply:
So, this is just the hidden module that enables to add system modules and module dependencies to a view.

The reason I am suggesting the change from `.view_module` for all views to `.view_<view_name>` relates to a scenario with multiple views in use at same time and Lmod.
E.g. suppose we have a set of MPI views (openmpi, mpich, ..) and a set of GPU views (nvidia, amd). I want to load 2 views, e.g. openmpi and nvidia, for whatever reason.
In Lmod you can only ask to load a modulefile by its name, not its full path; Lmod will look at MODULEPATH in order, and load the first modulefile it find with that name. So if you are loading TWO views, both calling the module named `.view_module`, the view that comes 2nd in the MODULEPATH will wrongly load `.view_module` from the 1st view. Hence we need to have view specific names, `.view_<view_name>`.

In TCL you can specify to load a modulefile by its full path, so there is no mistake. I have just implemented the same thing for consistency with Lmod.